tests/e2e/Services/Functions/FunctionsCustomServerTest.php

```diff
@@ -303,6 +299,41 @@ public function testCreateDeployment($data):array
 
         // Wait for deployment to build.
         sleep(15);
+
+        /**
+         * Test for Large Code File SUCCESS
+         */
+        $source = realpath(__DIR__ . '/../../../resources/functions/php-large.tar.gz');
+        $chunkSize = 5*1024*1024;
+        $handle = @fopen($source, "rb");
+        $mimeType = 'application/x-gzip';
+        $counter = 0;
+        $size = filesize($source);
+        $headers = [
+            'content-type' => 'multipart/form-data',
+            'x-appwrite-project' => $this->getProject()['$id']
+        ];
+        $id = '';
+        while (!feof($handle)) {
+            $curlFile = new \CURLFile('data://' . $mimeType . ';base64,' . base64_encode(@fread($handle, $chunkSize)), $mimeType, 'php-large-fx.tar.gz');
+            $headers['content-range'] = 'bytes ' . ($counter * $chunkSize) . '-' . min(((($counter * $chunkSize) + $chunkSize) - 1), $size) . '/' . $size;
+            if(!empty($id)) {
+                $headers['x-appwrite-id'] = $id;
+            }
+            $largeTag = $this->client->call(Client::METHOD_POST, '/functions/'.$data['functionId'].'/deployments', array_merge($headers, $this->getHeaders()), [
+                'entrypoint' => 'index.php',
+                'code' => $curlFile,
+            ]);
+            $counter++;
+            $id = $largeTag['body']['$id'];
+        }
+        @fclose($handle);
+
+        $this->assertEquals(201, $largeTag['headers']['status-code']);
+        $this->assertNotEmpty($largeTag['body']['$id']);
+        $this->assertIsInt($largeTag['body']['dateCreated']);
+        $this->assertEquals('index.php', $largeTag['body']['entrypoint']);
+        $this->assertGreaterThan(10000, $largeTag['body']['size']);
```

Comment:
Also, let's keep the large file upload test, as an independent test and not couple it with the existing `createDeployment` test. 

Let's do something like the `testCreateCustomNodeExecution()` method . This creates a deployment, an execution and even cleans up the container after the test. 

https://github.com/appwrite/appwrite/blob/feat-functions-refactor/tests/e2e/Services/Functions/FunctionsCustomServerTest.php#L851